pkg/logger/logger.go

```diff
@@ -0,0 +1,139 @@
+package logger
+
+import (
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+/*
+This package wraps the zap logging API for use across services without passing a shared logger.
+Fields can be added to a requestID using the WriteFields method, these will be added to any
+subsequent XxxWithID log calls. To preserve performance ClearFields must be called when the
+requestID's thread is closed as a sync map is used internally.
+Child loggers can be created from a parent logger using the WithFields method, this child logger
+will append the provided fields to all logs, whilst maintaining a reference to the top level
+request fields pool.
+
+Example:
+
+WrappedLogger := NewLogger(myLogger)
+WrappedLogger.WriteFields("my-id", zap.String("foo", "bar"))
+WrappedLogger2 := WrappedLogger.WithFields(zap.String("ping", "pong"))
+
+WrappedLogger.DebugWithID("my-id", "my log line")
+	=> {"level":"debug","foo":"bar","msg":"my log line""}
+
+WrappedLogger2.DebugWithID("my-id", "my log line")
+	=> {"level":"debug","foo":"bar","ping":"pong","msg":"my log line""}
+
+WrappedLogger2.WriteFields("my-id", zap.String("food", "bars"))
+
+WrappedLogger.DebugWithID("my-id", "my log line")
+	=> {"level":"debug","foo":"bar","food":"bars","msg":"my log line""}
+*/
+
+type Logger struct {
+	requestFields *sync.Map
+	Logger        *zap.Logger
+	fields        []zap.Field
+}
+
+func (l *Logger) DebugWithID(reqID string, msg string, fields ...zap.Field) {
+	l.Logger.Debug(msg, l.getFieldsForLog(reqID)...)
+}
+
+func (l *Logger) Debug(msg string, fields ...zap.Field) {
+	fields = append(fields, l.fields...)
+	l.Logger.Debug(msg, fields...)
+}
+
+func (l *Logger) InfoWithID(reqID string, msg string, fields ...zap.Field) {
+	l.Logger.Info(msg, l.getFieldsForLog(reqID)...)
```

Comment:
The `fields` passed into the func aren't used (same for other similar funcs)

Reply:
resolved